python/ray/air/checkpoint.py

```diff
@@ -454,7 +454,8 @@ def from_directory(cls, path: str) -> "Checkpoint":
 
         return checkpoint
 
-    @classmethod
+    @classmethod  
+    # TODO: Deprecate `from_checkpoint`. For context, see #29058.
```

Comment:
If this is something that's already aligned on and there is an alternative and better solution, should we go ahead and mark this as `@Deprecated`?

Reply:
Followed up on this offline -- we can't deprecate this until #29857 is fixed

Reply:
Can you update https://github.com/ray-project/ray/issues/29058 to reflect that? It currently says:
> from_checkpoint is a hack. Now that we've addressed the underlying issue, we should remove it.

and

> So, Checkpoint.from_checkpoint is now obsolete.

Reply:
Thanks. Update the issue.